packages/flutter/lib/src/rendering/box.dart

```diff
@@ -209,6 +209,46 @@ class BoxConstraints extends Constraints {
     return result;
   }
 
+  /// Returns a size that attempts to meet the following conditions, in order:
+  ///
+  ///  - The size must satisfy the constraints.
+  ///  - The aspect ratio of the returned size matches the aspect ratio of the
+  ///    given size.
+  ///  - The retruned size is maximal subject to being smaller than the given
```

Comment:
Also I don't understand bullet three. Do you mean "The returned size as big as possible while still being equal to or smaller than the given size"?